CITY/src/City/Networks/CityMain.cs

```diff
@@ -0,0 +1,247 @@
+         Base58Prefixes[(int)Base58Type.CONFIRMATION_CODE] = new byte[] { 0x64, 0x3B, 0xF6, 0xA8, 0x9A };
+         Base58Prefixes[(int)Base58Type.STEALTH_ADDRESS] = new byte[] { 0x2a };
+         Base58Prefixes[(int)Base58Type.ASSET_ID] = new byte[] { 23 };
+         Base58Prefixes[(int)Base58Type.COLORED_ADDRESS] = new byte[] { 0x13 };
+
+         Bech32Encoders = new Bech32Encoder[2];
+         var encoder = new Bech32Encoder(CitySetup.Main.CoinTicker);
+         Bech32Encoders[(int)Bech32Type.WITNESS_PUBKEY_ADDRESS] = encoder;
+         Bech32Encoders[(int)Bech32Type.WITNESS_SCRIPT_ADDRESS] = encoder;
+
+         Checkpoints = CitySetup.Main.Checkpoints;
+         DNSSeeds = CitySetup.Main.DNS;
+         SeedNodes = CitySetup.Main.Nodes;
+
+         StandardScriptsRegistry = new CityStandardScriptsRegistry();
+
+         // 64 below should be changed to TargetSpacingSeconds when we move that field.
+         Assert(DefaultBanTimeSeconds <= Consensus.MaxReorgLength * 64 / 2);
+
+         Assert(Consensus.HashGenesisBlock == uint256.Parse(CitySetup.Main.HashGenesisBlock));
+         Assert(Genesis.Header.HashMerkleRoot == uint256.Parse(CitySetup.Main.HashMerkleRoot));
+
+         RegisterRules(Consensus);
+         RegisterMempoolRules(Consensus);
+      }
+
+      protected void RegisterRules(IConsensus consensus)
+      {
+         consensus.ConsensusRules
+             .Register<HeaderTimeChecksRule>()
+             .Register<HeaderTimeChecksPosRule>()
+             .Register<PosFutureDriftRule>()
+             .Register<CheckDifficultyPosRule>()
+             .Register<CityHeaderVersionRule>()
+             .Register<ProvenHeaderSizeRule>()
+             .Register<ProvenHeaderCoinstakeRule>();
+
+         consensus.ConsensusRules
+             .Register<BlockMerkleRootRule>()
+             .Register<PosBlockSignatureRepresentationRule>()
+             .Register<PosBlockSignatureRule>();
+
+         consensus.ConsensusRules
+             .Register<SetActivationDeploymentsPartialValidationRule>()
+             .Register<PosTimeMaskRule>()
+
+             // rules that are inside the method ContextualCheckBlock
+             .Register<TransactionLocktimeActivationRule>()
+             .Register<CoinbaseHeightActivationRule>()
+             .Register<WitnessCommitmentsRule>()
+             .Register<BlockSizeRule>()
+
+             // rules that are inside the method CheckBlock
+             .Register<EnsureCoinbaseRule>()
+             .Register<CheckPowTransactionRule>()
+             .Register<CheckPosTransactionRule>()
+             .Register<CheckSigOpsRule>()
+             .Register<PosCoinstakeRule>();
+
+         consensus.ConsensusRules
+             .Register<SetActivationDeploymentsFullValidationRule>()
+
+             .Register<CheckDifficultyHybridRule>()
+
+             // rules that require the store to be loaded (coinview)
+             //.Register<FetchUtxosetRule>()
+             .Register<LoadCoinviewRule>()
```

Comment:
Why not use the new rule?

Reply:
Not tested and verified with it, and our samples doesn't yet use the new rules. If you update the sample, then next time we generate this codebase it will use the new rules. I'm hoping we can most of the time just re-generate the source code when we update these reference nodes.

Reply:
ok I will do that yes I will modify the samples template